### PR TITLE
Store created date/time

### PR DESCRIPTION
**Pull request by edurenye**
This module stores the last accessed time for URL redirects. It would sure be nice for it to store the creation time, as well.
Related to this issue: https://www.drupal.org/node/1853144

### CODE REVIEW

File: src/Controller/RedirectListBuilder.php [hunk 1]
```diff
@@ -33,6 +34,7 @@ public function buildRow(EntityInterface $redirect) {
       $row['redirect_redirect']['data'] = '';
     }
     $row['status_code']['data'] = $redirect->getStatusCode();
+    $row['creation']['data'] = format_date($redirect->getCreationDateTime(), 'short');
```
Comment (LKS90) [inline src/Controller/RedirectListBuilder.php]: `Drupal::service('date.formatter')->format($redirect->getCreationDateTime(), 'short')` is the Drupal 8 way of doing that. 

the function you use basically just does this:

```
 function format_date($timestamp, $type = 'medium', $format = '', $timezone = NULL, $langcode = NULL) {
    return \Drupal::service('date.formatter')->format($timestamp, $type, $format, $timezone, $langcode);
 }
```

**Comment by LKS90** [comment]
It's just Travis CI acting out, branch should pass tests:

https://travis-ci.org/md-systems/redirect/jobs/71375494

File: redirect.module [hunk 2]
```diff
@@ -88,6 +88,7 @@ function redirect_path_update(array $path) {
       $redirect->setRedirect($path['source']);
       $redirect->setLanguage($original_path['langcode']);
       $redirect->setStatusCode(\Drupal::config('redirect.settings')->get('default_status_code'));
+      $redirect->setCreationDateTime(REQUEST_TIME);
```
Comment (Berdir) [inline redirect.module]: this should not be needed.

File: src/Form/RedirectForm.php [hunk 3]
```diff
@@ -60,6 +60,7 @@ protected function prepareEntity() {
       }
 
       $redirect->setLanguage($this->getRequest()->get('language') ? $this->getRequest()->get('language') : Language::LANGCODE_NOT_SPECIFIED);
+      $redirect->setCreationDateTime(REQUEST_TIME);
```
Comment (Berdir) [inline src/Form/RedirectForm.php]: Same here. A created field type automatically is set to this value.

File: src/Entity/Redirect.php [hunk 4]
```diff
@@ -320,6 +340,9 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setDescription(t('The redirect status code.'))
       ->setDefaultValue(0);
 
+    $fields['creation'] = BaseFieldDefinition::create('created')
```
Comment (Berdir) [inline src/Entity/Redirect.php]: The field is called created everywhere else, let's use that here as well.

**Comment by Berdir** [comment]
Merged.

**Pull request closed without merging**